src/main/java/gregtech/api/unification/OreDictUnifier.java

```diff
@@ -267,21 +275,40 @@ else if ((materialAmount * 9) >= M)
         return ItemStack.EMPTY;
     }
 
-
     public static ItemStack getDust(MaterialStack materialStack) {
         return getDust(materialStack.material, materialStack.amount);
     }
 
     public static ItemStack getIngot(Material material, long materialAmount) {
         if (!material.hasProperty(PropertyKey.INGOT) || materialAmount <= 0)
             return ItemStack.EMPTY;
+        if (materialAmount % (M * 9) == 0)
+            return get(OrePrefix.block, material, (int) (materialAmount / (M * 9)));
         if (materialAmount % M == 0 || materialAmount >= M * 16)
             return get(OrePrefix.ingot, material, (int) (materialAmount / M));
         else if ((materialAmount * 9) >= M)
             return get(OrePrefix.nugget, material, (int) ((materialAmount * 9) / M));
         return ItemStack.EMPTY;
     }
 
+    public static ItemStack getIngot(MaterialStack materialStack) {
+        return getIngot(materialStack.material, materialStack.amount);
+    }
+
+    /**
+     * Returns an Ingot of the material if it exists. Otherwise it returns a Dust.
+     * Returns ItemStack.EMPTY if neither exist.
+     */
+    public static ItemStack getIngotOrDust(Material material, long materialAmount) {
+        ItemStack ingotStack = getIngot(material, materialAmount);
+        if (ingotStack != ItemStack.EMPTY) return ingotStack;
+        return getDust(material, materialAmount);
+    }
+
+    public static ItemStack getIngotOrDust(MaterialStack materialStack) {
+        return getIngotOrDust(materialStack.material, materialStack.amount);
+    }
```

Comment:
Why did you add this method, when anything that can call this method can automatically call the method's redirect, as it has access to the `materialStack.amount` and `materialStack.material` already. This seems like a useless method